### PR TITLE
Update bind-mounts.md

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
### Proposed changes
A warning about mount propagation will be helpful in the docs. See https://github.com/kubernetes-sigs/kind/issues/2576 for details.

### Related issues (optional)
Related to https://github.com/kubernetes-sigs/kind/issues/2576

### CODE REVIEW

**Comment by NikhilSharmaWe** [comment]
@usha-mandya Could you please review the warning added in the docs about mount propagation.

**Comment by NikhilSharmaWe** [comment]
@tonistiigi Could you please review the warning added in the docs about mount propagation.

File: storage/bind-mounts.md [hunk 1]
```diff
@@ -289,6 +289,9 @@ propagation setting has a recursive counterpoint. In the case of recursion,
 consider that `/tmp/a` is also mounted as `/foo`. The propagation settings
 control whether `/mnt/a` and/or `/tmp/a` would exist.
 
+Warning : mount propagation will not work with the docker desktop app, this is a Docker limitation. See
+> [kubernetes-sigs/kind #2576](https://github.com/kubernetes-sigs/kind/issues/2576) for details.
+
```
Comment (NikhilSharmaWe) [inline storage/bind-mounts.md:294]: @tonistiigi Could you please review the warning added related to mount propogation.

File: storage/bind-mounts.md [hunk 2]
```diff
@@ -289,6 +289,9 @@ propagation setting has a recursive counterpoint. In the case of recursion,
 consider that `/tmp/a` is also mounted as `/foo`. The propagation settings
 control whether `/mnt/a` and/or `/tmp/a` would exist.
 
+Warning : mount propagation will not work with the docker desktop app, this is a Docker limitation. See
```
Comment (usha-mandya) [inline storage/bind-mounts.md:292]:
```suggestion
> **Warning:** 
> 
> Mount propagation doesn't work with the Docker Desktop.
{: .warning
```

File: storage/bind-mounts.md [hunk 3]
```diff
@@ -289,6 +289,9 @@ propagation setting has a recursive counterpoint. In the case of recursion,
...
+> [kubernetes-sigs/kind #2576](https://github.com/kubernetes-sigs/kind/issues/2576) for details.
```
Comment (usha-mandya) [inline storage/bind-mounts.md:293]:
```suggestion
```

File: storage/bind-mounts.md [hunk 4]
```diff
@@ -289,6 +289,11 @@ propagation setting has a recursive counterpoint. In the case of recursion,
 consider that `/tmp/a` is also mounted as `/foo`. The propagation settings
 control whether `/mnt/a` and/or `/tmp/a` would exist.
 
+> **Warning:** 
+> 
+> Mount propagation doesn't work with the Docker Desktop.
```
Comment (usha-mandya) [inline storage/bind-mounts.md:294]:
```suggestion
> Mount propagation doesn't work with Docker Desktop.
```

**Review by usha-mandya — APPROVED** [review]

**Pull request merged**